clients/drcachesim/tools/view.cpp

```diff
@@ -342,33 +358,57 @@ view_t::process_memref(const memref_t &memref)
         return true;
     }
 
+    static constexpr int name_width = 12;
     if (!type_is_instr(memref.instr.type) &&
         memref.data.type != TRACE_TYPE_INSTR_NO_FETCH) {
-        // XXX: Print prefetch info.
+        std::string name; // Shared output for address-containing types.
         switch (memref.data.type) {
-        case TRACE_TYPE_READ:
-            std::cerr << "read   " << std::setw(2) << memref.data.size << " byte(s) @ 0x"
-                      << std::hex << std::setfill('0') << std::setw(sizeof(void *) * 2)
-                      << memref.data.addr << " by PC 0x" << std::setw(sizeof(void *) * 2)
-                      << memref.data.pc << std::dec << std::setfill(' ') << "\n";
-            break;
-        case TRACE_TYPE_WRITE:
-            std::cerr << "write  " << std::setw(2) << memref.data.size << " byte(s) @ 0x"
-                      << std::hex << std::setfill('0') << std::setw(sizeof(void *) * 2)
-                      << memref.data.addr << " by PC 0x" << std::setw(sizeof(void *) * 2)
-                      << memref.data.pc << std::dec << std::setfill(' ') << "\n";
-            break;
+        default: std::cerr << "<entry type " << memref.data.type << ">\n"; return true;
         case TRACE_TYPE_THREAD_EXIT:
             std::cerr << "<thread " << memref.data.tid << " exited>\n";
-            break;
-        default: std::cerr << "<entry type " << memref.data.type << ">\n"; break;
+            return true;
+            // The rest are address-containing types.
+        case TRACE_TYPE_READ: name = "read"; break;
+        case TRACE_TYPE_WRITE: name = "write"; break;
+        case TRACE_TYPE_INSTR_FLUSH: name = "iflush"; break;
+        case TRACE_TYPE_DATA_FLUSH: name = "dflush"; break;
+        case TRACE_TYPE_PREFETCH: name = "pref"; break;
+        case TRACE_TYPE_PREFETCH_READ_L1: name = "pref-L1"; break;
```

Comment:
s/pref-L1/pref-r-L1/

and also for the next two.

Reply:
Done.